src/main/java/io/debezium/connector/spanner/config/validation/ConnectionValidator.java

```
@@ -56,10 +56,7 @@ public ConnectionValidator validate() {
         if (!FieldValidator.isSpecified(googleCredentials) && !FieldValidator.isSpecified(credentialPath) && !FieldValidator.isSpecified(credentialJson)) {
             String message = String.format(PLEASE_SPECIFY_CONFIGURATION_PROPERTY_MSG, SPANNER_CREDENTIALS_PATH.name(),
                     SPANNER_CREDENTIALS_JSON.name(), GOOGLE_APPLICATION_CREDENTIALS_ENV_VAR);
-            LOGGER.error(message);
-            context.error(message, SPANNER_CREDENTIALS_PATH, SPANNER_CREDENTIALS_JSON);
-            result = false;
-            return this;
+            LOGGER.warn(message, SPANNER_CREDENTIALS_PATH, SPANNER_CREDENTIALS_JSON);
```

It should not be warning, just `INFO`.
Also the validation should be kept there but modified. If any of the settings is set but not all then there should be validation error. If none is set then the App Default Credentials will be used.

---

Fixed: https://github.com/debezium/debezium-connector-spanner/pull/65/commits/678725a918c7c947a8fba9d223795b82c2072f69

---

I'm wondering if we should add validation here that if there is no credentials JSON / path specified, there should be application default credentials specified. If not, we should throw an error.

Then, we can probably add a unittest here?

https://github.com/debezium/debezium-connector-spanner/blob/9306f1349643cf1d1c56ed1c335615a5f916cb2e/src/test/java/io/debezium/connector/spanner/config/validation/ConnectionValidatorTest.java#L22

---

https://github.com/debezium/debezium-connector-spanner/blob/9306f1349643cf1d1c56ed1c335615a5f916cb2e/src/test/java/io/debezium/connector/spanner/config/validation/ConnectionValidatorTest.java#L38
This test case is Disabled. Since the validation has been changed to just check the key and output the INFO log, I don't see the need to change the current test.